packages/blockchain-extension/extension/commands/packageSmartContractCommand.ts

```diff
@@ -249,9 +301,31 @@ async function javaPackageAndVersion(overrideName?: string, overrideVersion?: st
 /**
  * Method which calls an input box should the project be coded in golang, which asks the user for a package name and version
  * (as golang projects do not contain a package.json file), and returns an object containing both these values.
+ * @param workspaceDir {String} workspaceDir A string containing the path to the current active workspace (the workspace of the project the user is packaging).
  * @returns {string, string} Returns an object with the workspacePackageName and workspacePackageVersion which will be used in the createPackageDir() method
  */
-async function golangPackageAndVersion(overrideName?: string, overrideVersion?: string): Promise<{ workspacePackageName: string, workspacePackageVersion: string }> {
+async function golangPackageAndVersion(workspaceDir: vscode.WorkspaceFolder, overrideName?: string, overrideVersion?: string): Promise<{ workspacePackageName: string, workspacePackageVersion: string }> {
+
+    // Check if go smart contract uses version greater that fabric v2
+
+    const workspaceMod: string = path.join(workspaceDir.uri.fsPath, '/go.mod');
+    let workspaceModContents: string;
+
+    if (await fs.pathExists(workspaceMod)) {
+        workspaceModContents = await fs.readFile(workspaceMod, 'utf8');
+        const regex: RegExp = new RegExp(/github\.com\/hyperledger\/fabric-contract-api-go /);
+        if (!(regex.test(workspaceModContents))) {
+            const regexChaincode: RegExp = new RegExp(/github\.com\/hyperledger\/fabric-chaincode-go /);
+            if (!(regexChaincode.test(workspaceModContents))) {
+                throw new Error(`In this version of the extension, you can only package contracts which use fabric v2 or greater`);
+            }
+        }
+    } else {
+        const warnPackage: boolean = await UserInputUtil.showConfirmationWarningMessage(`Could not find the fabric-contract-api-go version. Would you like to package it anyway?`);
```

Comment:
`Could not detect SDK version. Would you like to package it anyway?` - we should probably have all of these prompts be consistent throughout the languages.